### DISCUSSION TITLE
Change Java source and target version

### DISCUSSION THREAD

**Discussion by ghost** [github/securitylab#142] [General] [answered]
The codeql extractor for java sets the source and the target flags for the `javac` compiler to 8.

I wish to import ```java.net.http.HttpClient``` but it was not present in Java 8. It was included only in Java version 9. I am using java 11 on my host and I can compile my code with it properly. 

So how do I change the source version for the codeql extractor to Java 11 here?

I tried adding `-source 11` in the options file but in vain.

**Comment by criemen** [comment, marked as answer]
Hi @porcupineyhairs,
I am really sorry for the late response.

I talked to the engineering team, and they assured me that this should work now.

Please let me know if you still have any questions!
